### PR TITLE
Add initial markup

### PR DESCRIPTION
**Pull request by asutptec4**

### CODE REVIEW

File: src/components/MovieCounter/MovieCounter.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const MovieCounter = (props) => {
+  return <div className="c-p-v-15">{props.movieCount} movies found</div>;
```
Comment (omelenetskiy) [inline src/components/MovieCounter/MovieCounter.js]: Strange naming `c-p-v-15`

File: src/components/MovieList/MovieList.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import MovieCard from '../MovieCard/MovieCard';
+import MovieCounter from '../MovieCounter/MovieCounter';
+import './MovieList.scss';
+
+const MovieList = (props) => {
```
Comment (omelenetskiy) [inline src/components/MovieList/MovieList.js]: lets extract props in arguments like
`const MovieList = ({ movies }) => {`

File: src/components/MovieListControl/MovieListControl.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import SelectOption from '../../utils/SelectOption';
+import Dropdown from '../Dropdown/Dropdown';
+import './MovieListControl.scss';
+
+const MovieListControl = (props) => {
```
Comment (omelenetskiy) [inline src/components/MovieListControl/MovieListControl.js]: the same here

File: src/components/MovieListControl/MovieListControl.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="movie-list-controls ">
+      <div className="filter-control">
+        {props.filterOptions.map((g) => (
```
Comment (omelenetskiy) [inline src/components/MovieListControl/MovieListControl.js]: let's set the default props === [] for `filterOptions`

File: src/components/MovieListControl/MovieListControl.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        <label htmlFor="sort-selector">Sort by</label>
+        <Dropdown options={props.sortOptions} id="sort-selector" />
+      </div>
+    </div>
+  );
+};
+
+MovieListControl.propTypes = {
+  filterOptions: PropTypes.arrayOf(SelectOption),
+  sortOptions: PropTypes.arrayOf(SelectOption),
```
Comment (omelenetskiy) [inline src/components/MovieListControl/MovieListControl.js]: isRequired or default props for sortOptions

File: src/containers/App/App.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';
+import Footer from '../Footer/Footer';
+import Header from '../Header/Header';
+import MovieListContainer from '../MovieListContainer/MovieListContainer';
+import './App.scss';
+
+function App() {
```
Comment (omelenetskiy) [inline src/containers/App/App.js]: `export const App ` or  `const App = () => `

File: src/containers/MovieListContainer/MovieListContainer.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  { id: 'hor', name: 'Horror' },
+  { id: 'cri', name: 'Crime' },
+];
+
+const sortOptions = [
+  { id: 'date', name: 'Release Date', selected: true },
+  { id: 'votes', name: 'Average Votes' },
+];
+
+const WithNoFound = (Component) => {
```
Comment (omelenetskiy) [inline src/containers/MovieListContainer/MovieListContainer.js]: I think it will be better to move outside this HOC

File: src/index.js [hunk 8]
```diff
@@ -1,7 +1,11 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-import { Greeting } from './Greeting';
-import './index.scss';
+import App from './containers/App/App';
 
-ReactDOM.render(<Greeting name="React" />, document.getElementById('root'));
+ReactDOM.render(
+  <React.StrictMode>
```
Comment (omelenetskiy) [inline src/index.js]: It is not needed to make it strict

File: src/utils/SelectOption.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+export default class SelectOption {
```
Comment (omelenetskiy) [inline src/utils/SelectOption.js]: Why do we need it?
let's discuss

**Comment by omelenetskiy** [comment]
I think you should add responsive possibilities to your mock-up
On mobile devices looks not good

File: src/components/MovieCounter/MovieCounter.js (hunk 1 above)
Comment (asutptec4) [reply src/components/MovieCounter/MovieCounter.js]: Updated in 4ab9d8b

File: src/components/MovieList/MovieList.js (hunk 2 above)
Comment (asutptec4) [reply src/components/MovieList/MovieList.js]: Looks great! Updated in 2b872ed

File: src/containers/MovieListContainer/MovieListContainer.js (hunk 7 above)
Comment (asutptec4) [reply src/containers/MovieListContainer/MovieListContainer.js]: Updated in 596b2c7

File: src/components/MovieListControl/MovieListControl.js (hunk 4 above)
Comment (asutptec4) [reply src/components/MovieListControl/MovieListControl.js]: Updated in 9bc0633

File: src/utils/SelectOption.js (hunk 9 above)
Comment (asutptec4) [reply src/utils/SelectOption.js]: I haven't used it yet. I would use common model for a few components and don't want to duplicate propTypes across them.
Where is a good place for that?

**Comment by asutptec4** [comment]
> I think you should add responsive possibilities to your mock-up
> On mobile devices looks not good

I thought about it. As I have no experience with it, do you have any objection if I do it as a part of the next task?

**Comment by omelenetskiy** [comment]
> > I think you should add responsive possibilities to your mock-up
> > On mobile devices looks not good
> 
> I thought about it. As I have no experience with it, do you have any objection if I do it as a part of the next task?

No objections from my side
as I remember we don't have any requirements to make it works on mobile as well
so, let's do it a bit later

File: src/utils/SelectOption.js (hunk 9 above)
Comment (omelenetskiy) [reply src/utils/SelectOption.js]: I believe we can try to use `centralized PropTypes` react pattern in this case 

please have a look an example 
https://www.freecodecamp.org/news/react-pattern-centralized-proptypes-f981ff672f3b/

What do you think?

**Review by omelenetskiy — APPROVED** [review]

**Comment by omelenetskiy** [comment]
PR approved
please merge it after the completing all fixes

**Comment by omelenetskiy** [comment]
Great job!

File: src/utils/SelectOption.js (hunk 9 above)
Comment (asutptec4) [reply src/utils/SelectOption.js]: Thank you for link! It is definitely what I need.

**Pull request merged**